### PR TITLE
[s]Prevents AI units from clicking on things they can't see

### PR DESCRIPTION
**Pull request by Markolie**
Ports over https://github.com/tgstation/tgstation/pull/21912 and https://github.com/tgstation/tgstation/pull/21957 from /tg/: it prevents AI's from being able to interact with turfs they're unable to see on their camera. This relates to the exploit where players can remove the AI static using Cheat Engine. 

This also ports over a bunch of useful admin shortcut defines that we should probably start using from here on out: it means we don't have to copy-paste admin topic commands.

### CODE REVIEW

**Comment by BiberDark** [comment]
So the Ai is unable to interact in rooms because someone broke the lights ? This is bad in my opinion. Only because there could possible one who use this hack, is not enough to strip the ai even of more power, then it already is. And no the camera light is not enough to handle this.

**Comment by Markolie** [comment]
No, this only applies to things the AI actually can't see. If they can see the APC, they can interact with it. Specifically, this is about the AI static that overlays in areas where there's no camera coverage.

The borderline: it doesn't change anything in the way AI's play and doesn't affect legit players. This solely prevents an exploit using Cheat Engine.

File: code/__DEFINES/admin.dm [hunk 1]
```diff
@@ -43,4 +43,21 @@
...
+#define ADMIN_BSA(user) "(<a href='?_src_=holder;BlueSpaceArtillery=\ref[user]'>BSA</a>)"
+#define ADMIN_CENTCOM_REPLY(user) "(<a href='?_src_=holder;CentcommReply=\ref[user]'>RPLY</a>)"
+#define ADMIN_SYNDICATE_REPLY(user) "(<a href='?_src_=holder;SyndicateReply=\ref[user]'>RPLY</a>)"
+#define ADMIN_SC(user) "(<a href='?_src_=holder;adminspawncookie=\ref[user]'>SC</a>)"
+#define ADMIN_LOOKUP(user) "[key_name_admin(user)][ADMIN_QUE(user)]"
+#define ADMIN_LOOKUPFLW(user) "[key_name_admin(user)][ADMIN_QUE(user)] [ADMIN_FLW(user)]"
+#define ADMIN_FULLMONTY(user) "[key_name_admin(user)] [ADMIN_QUE(user)] [ADMIN_PP(user)] [ADMIN_VV(user)] [ADMIN_SM(user)] [ADMIN_FLW(user)] [ADMIN_TP(user)]"
+#define ADMIN_JMP(src) "(<a href='?_src_=holder;adminplayerobservecoodjump=1;X=[src.x];Y=[src.y];Z=[src.z]'>JMP</a>)"
+#define COORD(src) "[src ? "([src.x],[src.y],[src.z])" : "nonexistent location"]"
+#define ADMIN_COORDJMP(src) "[src ? "[COORD(src)] [ADMIN_JMP(src)]" : "nonexistent location"]"
```
Comment (Crazylemon64) [inline code/__DEFINES/admin.dm]: Oh, these will be handy elsewhere :)

File: code/__DEFINES/admin.dm [hunk 2]
```diff
@@ -43,4 +43,21 @@
 
 #define R_MAXPERMISSION 32768 //This holds the maximum value for a permission. It is used in iteration, so keep it updated.
 
-#define R_HOST			65535
\ No newline at end of file
+#define R_HOST			65535
+
+#define ADMIN_QUE(user) "(<a href='?_src_=holder;adminmoreinfo=\ref[user]'>?</a>)"
```
Comment (Fox-McCloud) [inline code/__DEFINES/admin.dm]: if I recall correctly we don't use \ref anymore, but UID.

**Comment by Earthykiller127** [comment]
Those are some pretty peculiar words for a fix that doesn't even work

**Comment by Markolie** [comment]
As far as I know, this fix works fine. I know there's workarounds for the static removal right now, but on the server side this at least prevents interaction with whatever is on the turf that's supposed to be static.

**Pull request merged**